compiler/next/include/chpl/uast/Reduce.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_REDUCE_H
+#define CHPL_UAST_REDUCE_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/queries/UniqueString.h"
+#include "chpl/uast/Call.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a reduction.
+*/
+class Reduce final : public Call {
+ private:
+  Reduce(ASTList children, UniqueString op)
+    : Call(asttags::Reduce, std::move(children),
+           /*hasCalledExpression*/ false),
+      op_(op) {
+    assert(numChildren() == 1);
+    assert(!op_.isEmpty());
+  }
+
+  bool contentsMatchInner(const ASTNode* other) const override {
+    const Reduce* rhs = other->toReduce();
```

Comment:
Is `other` guaranteed to be a `Reduce`?

Reply:
yes, that is checked before this function is called

Reply:
Yes, I think so.